aii-ks/src/main/perl/ks.pm

```diff
@@ -408,16 +408,8 @@ sub kscommands
     my @packages = @{$tree->{packages}};
     push(@packages, 'bind-utils'); # required for nslookup usage in ks-post-install
 
-    my $installtype = $tree->{installtype};
-    if ($installtype =~ /http/) {
-        my ($proxyhost, $proxyport, $proxytype) = proxy($config);
-        if ($proxyhost && $proxytype eq "reverse") {
-            if ($proxyport) {
-                $proxyhost .= ":$proxyport";
-            }
-            $installtype =~ s{(https?)://([^/]*)/}{$1://$proxyhost/};
-        }
-    }
+    my $proxy_config = [proxy($config)];
```

Comment:
Passing the information down as a bare list makes it more difficult to follow what's going on. Could you make `sub proxy()` return a hashref instead? Then you could write `$proxy_config->{host}` below, which would look so much nicer :-)

Reply:
done